src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -638,10 +641,30 @@ internal override bool IsRecord
 #nullable enable
         internal sealed override bool IsExtension => TypeKind == TypeKind.Extension;
 
-        internal sealed override TypeSymbol? ExtensionUnderlyingTypeNoUseSiteDiagnostics
+        internal sealed override bool IsExplicitExtension
         {
             get
             {
+                if (this.TypeKind != TypeKind.Extension)
+                    return false;
+
+                var uncommon = GetUncommonProperties();
+                Debug.Assert(uncommon != s_noUncommonProperties);
+
+                if (!uncommon.lazyIsExplicitExtension.HasValue())
+                {
+                    EnsureExtensionTypeDecoded(uncommon);
+                }
+
+                return uncommon.lazyIsExplicitExtension.Value();
+            }
+        }
+
+        internal sealed override TypeSymbol? ExtendedTypeNoUseSiteDiagnostics
+        {
+            get
+            {
+                // No cycles are possible because the extended type cannot be an extension type.
```

Comment:
>// No cycles are possible because the extended type cannot be an extension type. [](http://example.com/codeflow?start=16&length=80)

Have you confirmed that?

Reply:
Yes. When loading PE, we prevent the extended type from being an extension type (see `IsRestrictedExtensionUnderlyingType` check in `DecodeExtensionType`). We also prevent that in source.